dart-parser/src/parser/comment.rs

```diff
@@ -0,0 +1,65 @@
+use std::str;
+
+use nom::{
+    branch::alt,
+    bytes::complete::{is_not, tag},
+    combinator::{cut, eof, recognize},
+    multi::fold_many0,
+    sequence::{preceded, terminated, tuple},
+    IResult, Parser,
+};
+
+use crate::{dart::comment::Comment, parser::common::*};
+
+pub fn comments(s: &str) -> IResult<&str, Comment> {
+    alt((
+        comment.map(Comment::SingleLine),
+        multi_line_comment.map(Comment::MultiLine),
+    ))(s)
+}
+
+fn comment(s: &str) -> IResult<&str, &str> {
+    recognize(tuple((tag("//"), is_not("\r\n"), alt((br, eof)))))(s)
+}
+
+fn multi_line_comment(s: &str) -> IResult<&str, &str> {
+    recognize(preceded(
+        tag("/*"),
+        cut(terminated(
+            recognize(fold_many0(
+                alt((is_not("/**/"), multi_line_comment)),
+                || {},
+                |_, _| {},
+            )),
+            tag("*/"),
+        )),
+    ))(s)
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    #[test]
+    fn comment_test() {
+        assert_eq!(comment("// A comment\nx"), Ok(("x", "// A comment\n")));
+    }
+
+    #[test]
+    fn comment_eof_test() {
+        assert_eq!(comment("// A comment"), Ok(("", "// A comment")));
+    }
+
+    #[test]
+    fn multi_line_comment_test() {
+        assert_eq!(
+            multi_line_comment("/* A comment */"),
+            Ok(("", "/* A comment */"))
+        );
+        assert_eq!(
+            multi_line_comment("/* A comment\n /* another comment */\n*/"),
+            Ok(("", "/* A comment\n /* another comment */\n*/"))
+        );
```

Comment:
I'd prefer to split this in two separate test-cases and name them

- `comment_multi_line_test`
- `comment_multi_line_nested_test`

Let's also rename the single-line comment parser test-cases to match the new (as per my suggestion above) name: `comment_single_line_test`.